### PR TITLE
Feature/project repository interface

### PR DESCRIPTION
**Pull request by Jazinho**

### CODE REVIEW

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ProjectRepositoryImpl implements ProjectRepository {
+
+    private final String PROJECTS_FILE = "src/main/resources/projects.json";
```
Comment (wojtas626) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java]: you should get that from getClass().getClassLoader().getResourceAsAStream() (something like that!)

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            Path path = Paths.get(PROJECTS_FILE);
+            byte[] encoded = Files.readAllBytes(path);
+            String json = new String(encoded, Charset.forName("UTF-8"));
+            ProjectsConfig config = new Gson().fromJson(json, ProjectsConfig.class);
+
+            config.getProjects().add(project);
+            String jsonToSave = new Gson().toJson(config).toString();
+            PrintWriter out = new PrintWriter(PROJECTS_FILE);
+            out.println(jsonToSave);
+            out.close();
```
Comment (wojtas626) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java]: try with resources!

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+    }
+
+    @Override
+    public Project getProject(String projectName) {
+        try {
+            byte[] encoded = Files.readAllBytes(Paths.get(PROJECTS_FILE));
+            String json = new String(encoded, Charset.forName("UTF-8"));
+            ProjectsConfig config = new Gson().fromJson(json, ProjectsConfig.class);
+            System.out.println(config);
```
Comment (wojtas626) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java]: WHAT IS THIS!? Logger should be here

File: src/test/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.Test;
+import pl.edu.agh.kis.timetracker.domain.Project;
+import java.util.LinkedList;
+
+public class ProjectRepositoryTest {
+
+
+    @Test
+    public void shouldSaveProjectToFile() {
+        String name = "dupa";
```
Comment (wojtas626) [inline src/test/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryTest.java]: When client see string like this it seems to me he might not be happy :(

File: src/test/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        Project project = new Project(name, new LinkedList<>());
+
+
```
Comment (wojtas626) [inline src/test/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryTest.java]: remove empty line

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepository.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package pl.edu.agh.kis.timetracker.repository;
+
+import pl.edu.agh.kis.timetracker.domain.Project;
+
+public interface ProjectRepository {
+
+    void saveProject(Project project);
+
+    Project getProject(String projectName);
```
Comment (matkluska) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepository.java]: return Optional here

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java (hunk 2 above)
Comment (matkluska) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java]: Use try-with-resources

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+            for(Project project: config.getProjects()){
```
Comment (matkluska) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java]: use StreamAPI

File: src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                if(project.getName().equals(projectName)){
+                    return project;
+                }
+            }
+        } catch(IOException e){
+            e.printStackTrace();
```
Comment (matkluska) [inline src/main/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryImpl.java]: Add logger

File: src/test/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryTest.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import pl.edu.agh.kis.timetracker.domain.Project;
+import java.util.LinkedList;
+
+public class ProjectRepositoryTest {
+
+
+    @Test
+    public void shouldSaveProjectToFile() {
```
Comment (matkluska) [inline src/test/java/pl/edu/agh/kis/timetracker/repository/ProjectRepositoryTest.java]: missing given section

**Pull request merged**